autoprecompiles/src/lib.rs

```diff
@@ -234,8 +227,10 @@ pub fn generate_precompile<T: FieldElement>(
                 let local_identities = machine
                     .constraints
                     .iter()
-                    .map(|expr| SymbolicConstraint {
-                        expr: expr.expr.clone(),
+                    .map(|expr| {
+                        let mut expr = expr.expr.clone();
+                        powdr::substitute_algebraic(&mut expr, &sub_map);
```

Comment:
I think we need to add the substitutions performed by `local_cols` above to the map, no? Otherwise the cols here won't be replaced by `w*`

Reply:
Yes, sorry, I realized that after I requested the revision hehe :sweat_smile: